cluster/src/shard_tables_cache.rs

```diff
@@ -55,6 +58,46 @@ impl ShardTablesCache {
             .unwrap()
             .insert_or_update(tables_of_shard)
     }
+
+    /// Try to insert a new table to the shard with a newer version.
+    ///
+    /// It will fail if:
+    ///  - the shard doesn't exist, or
+    ///  - the shard version in the cache is not equal to the provided
+    ///    `prev_shard_version`, or
+    ///  - the table already exists.
+    pub fn try_insert_table_to_shard(
+        &self,
+        prev_shard_version: u64,
+        curr_shard: ShardInfo,
+        new_table: TableInfo,
+    ) -> Result<()> {
+        self.inner.write().unwrap().try_insert_table_to_shard(
+            prev_shard_version,
+            curr_shard,
+            new_table,
+        )
+    }
+
+    /// Try to remove a table from the shard with a newer version.
+    ///
+    /// It will fail if:
```

Comment:
The fail situations are same as insert_table_to_shard, how about: 
`It will fail in the same situations as insert_table_to_shard`

Reply:
Actually, there are some difference between the two cases, so let's keep it.

Reply:
Ok.